assets/js/HashStore.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+
+import {ReduceStore} from 'flux/utils';
+import Dispatcher from './Dispatcher';
+import Immutable from 'immutable';
+
+import type {Action} from './Actions';
+
+type State = Immutable.OrderedMap<string, number>;
+
+class HashStore extends ReduceStore<State> {
+
+    getInitialState(): State {
+        return Immutable.OrderedMap();
+    }
+
+    reduce (state: State, action: Action): State {
+        switch (action.type) {
+            case 'hash/bytesHashed':
+                return state.set(action.path, action.bytesHashed)
+            default:
+                return state;
+        }
+    }
+
+}
+
+// Export a singleton instance of the store, could do this some other way if
+// you want to avoid singletons.
```

Comment:
I'm inclined to say we could resonably use a singleton until a concrete use-case shows up saying otherwise but it might be worth moving this into the `BaggerApp` initialization since there aren't that many references and it'd be handy if we ever did need to support multiple instances.